source/common/http/http3/quic_client_connection_factory.h

```diff
@@ -10,17 +10,28 @@
 namespace Envoy {
 namespace Http {
 
+// Store quic helpers which can be shared between connections and must live
+// beyond the lifetime of individual connections.
+struct PersistentQuicInfo {
+  virtual ~PersistentQuicInfo() {}
+};
+
 // A factory to create EnvoyQuicClientSession and EnvoyQuicClientConnection for QUIC
 class QuicClientConnectionFactory : public Config::UntypedFactory {
 public:
   ~QuicClientConnectionFactory() override = default;
 
-  virtual std::unique_ptr<Network::ClientConnection>
-  createQuicNetworkConnection(Network::Address::InstanceConstSharedPtr server_addr,
-                              Network::Address::InstanceConstSharedPtr local_addr,
+  // Allows thread-local creation of PersistentQuicInfo.
```

Comment:
I can understand this object is per-conn pool. But is it thread-local? 
Plz add comment how to use these interfaces to create a client connection.

Reply:
connection pools must be thread local (they're not thread safe).  will comment :-)